usefulUtilities/loudApp/appResources/loudApp_app.js

```diff
@@ -0,0 +1,559 @@
+        AvatarList.avatarRemovedEvent.connect(removeUser);
+        Users.usernameFromIDReply.connect(setUserName);
+
+        Script.scriptEnding.connect(unload);
+
+        startUpdateInterval();
+    }
+
+
+    // Open the app callback
+    function onOpened() {
+        Users.requestsDomainListData = true;
+        startUpdateInterval();
+        addAllOverlays();
+    }
+
+
+    // Close the app callback
+    function onClosed() {
+        Users.requestsDomainListData = false;
+        stopUpdateInterval();
+        removeAllOverlays();
+    }
+
+
+    // Update UI
+    function doUIUpdate(key) {
+        ui.sendToHtml({
+            type: UPDATE_UI,
+            value: key ? settings[key] : settings,
+            key: key ? key : ""
+        });
+    }
+
+
+    // Sorts users by loudness in userStore then populates ten loudest list
+    var SEARCH_RADIUS_DEFAULT = 10; // radius that avatars are factored in muting and top 10 loudest
+    function sortAvatarsByLoudness() {
+        // sort by audioLevel
+        function sortNumber(a, b) {
+            return b.avgAudioLevel - a.avgAudioLevel;
+        }
+        removeAllOverlays();
+        var avatarList = settings.ui.searchWholeDomainForLoudestEnabled ? Object.keys(userStore) : getAvatarsInRadius(SEARCH_RADIUS_DEFAULT);
+        settings.users = avatarList.map(function (uuid) { return userStore[uuid]; });
+        settings.users = settings.users.sort(sortNumber).slice(0, 10);
+        addAllOverlays();
+    }
+
+
+    // Unloads the app
+    function unload() {
+        if (settings.users) {
+            settings.users.forEach(function(user) {
+                if (user.overlayID) {
+                    deleteOverlay(user.uuid)
+                }
+            });
+        }
+
+        stopUpdateInterval();
+        stopListening();
+
+        Users.usernameFromIDReply.disconnect(setUserName);
+        AvatarList.avatarAddedEvent.disconnect(addUser);
+        AvatarList.avatarRemovedEvent.disconnect(removeUser);
+    }
+
+    // #endregion APP
+
+
+    // #region OVERLAYS
+
+    // Adds overlay to users in top loudest list
+    var selectedUserUUID, // selected avatar with yellow overlay
+        OVERLAY_DEFAULT_DIMENSIONS = { x: 0.3, y: 0.3, z: 0.3 },
+        OVERLAY_DEFAULT_COLOR = { red: 255, blue: 255, green: 255 },
+        OVERLAY_SELECTED_COLOR = { red: 255, blue: 0, green: 255 };
+    function addOverlayToUser(uuid) {
+        var overlayPosition = AvatarList.getAvatar(uuid).getNeckPosition(); // user.currentPosition
+        var overlayProperties = {
+            position: overlayPosition, // assigned on creation
+            dimensions: OVERLAY_DEFAULT_DIMENSIONS,
+            solid: true,
+            parentID: uuid, // assigned on creation
+            color: OVERLAY_DEFAULT_COLOR,
+            drawInFront: true
+        };
+        var overlayID = Overlays.addOverlay("sphere", overlayProperties);
+        userStore[uuid].overlayID = overlayID;
```

Comment:
to simplify, you could do this (then delete the code on line 246):
```suggestion
        userStore[uuid].overlayID = Overlays.addOverlay("sphere", overlayProperties);
```

Reply:
Got it!